src/vdc_collection.rs

```diff
@@ -0,0 +1,134 @@
+use super::storage_manager::*;
+use serde_derive::{Deserialize, Serialize};
+use tracing::{info, info_span};
+use uuid::Uuid;
+
+/// Supported credential formats.
+#[derive(PartialEq, Debug, Serialize, Deserialize)]
+pub enum CredentialFormat {
+    MsoMdoc,
+    JwtVcJson,
+    LdpVc,
+    Other(String), // For ease of expansion.
+}
+
+/// Supported credential types.
+#[derive(PartialEq, Debug, Serialize, Deserialize)]
+pub enum CredentialType {
+    Iso18013_5_1mDl,
+    VehicleTitle,
+    Other(String), // For ease of expansion.
+}
+
+/// An individual credential.
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Credential {
+    id: Uuid,
+    format: CredentialFormat,
+    ctype: CredentialType,
+    payload: Vec<u8>, // The actual credential.
+}
+
+impl Credential {
+    /// Create a new credential.
+    fn new(
+        id: Uuid,
+        format: CredentialFormat,
+        ctype: CredentialType,
+        payload: Vec<u8>,
+    ) -> Credential {
+        Credential {
+            id,
+            format,
+            ctype,
+            payload,
+        }
+    }
+}
+
+/// Verifiable Digital Credential Collection
+///
+/// This is the main interface to credentials.
+pub struct VdcCollection {
+    storage: Box<dyn StorageManagerInterface>,
+}
+
+pub enum VdcCollectionError {
+    SerializeFailed,
+    DeserializeFailed,
+    StoreFailed,
+    LoadFailed,
+}
+
+impl VdcCollection {
+    /// Create a new credential set.
+    pub fn new(engine: Box<dyn StorageManagerInterface>) -> VdcCollection {
+        VdcCollection { storage: engine }
+    }
+
+    /// Add a credential to the set.
+    pub fn add(
+        &self,
+        id: Uuid,
+        format: CredentialFormat,
+        ctype: CredentialType,
+        payload: Vec<u8>,
+    ) -> Result<(), VdcCollectionError> {
+        let val = match serde_cbor::to_vec(&Credential::new(id, format, ctype, payload)) {
+            Ok(x) => x,
+            Err(_) => return Err(VdcCollectionError::SerializeFailed),
+        };
+
+        match self.storage.add(Key(id.to_string()), Value(val)) {
```

Comment:
I think it would be prudent to use a prefix to distinguish credentials from other files/objects that could be stored.

Reply:
I'll do that.

Reply:
Done.